kartothek/core/dataset.py

```diff
@@ -378,20 +378,26 @@ def get_indices_as_dataframe(
         Parameters
         ----------
         """
+        if not self.primary_indices_loaded:
+            # self.load_partition_indices is not inplace
+            dm = self.load_partition_indices()
+        else:
+            dm = self
+
         if columns is None:
-            columns = sorted(self.indices.keys())
-        elif columns == []:
-            return pd.DataFrame(index=self.partitions)
+            columns = sorted(dm.indices.keys())
+
+        if columns == []:
+            return pd.DataFrame(index=dm.partitions)
```

Comment:
Can you try putting these `if` blocks above the `         if not self.primary_indices_loaded:` block?

Reply:
I don't think I understand. The calls `dm.indices.keys()` and `dm.partitions` would not work if indices are not loaded. This is why I added the `dm = self.load_partition_indices` in the first place.

Reply:
hm, right. The issue with the `test_collect_dataset_metadata_empty_dataset` failure is related to calling this function with `columns=[]`. If you place another
```
 if columns == []:
            return pd.DataFrame(index=dm.partitions)
```
at the beginning, this will pass, but it's not the most elegant solution

Reply:
I added a commit on top to only load indices if `columns!=[]`.